src/hotel-management-app/src/app/restaurant/sales-analytics/sales-analytics.component.ts

```diff
@@ -0,0 +1,118 @@
+import { Component, OnInit } from '@angular/core';
+import { SalesService } from '../../services/sales.service';
+import { ActivatedRoute } from '@angular/router';
+import { Location } from '@angular/common';
+import { Sale } from '../../models/salesInfo.model';
+import { ChartData , ChartOptions } from '../../models/charts.model';
+import {Chart , registerables} from 'chart.js'
+Chart.register(...registerables)
+
+
+
+@Component({
+  selector: 'app-sales-analytics',
+  templateUrl: './sales-analytics.component.html',
+  styleUrls: ['./sales-analytics.component.css']
+})
+export class SalesAnalyticsComponent implements OnInit {
+  sales: Sale[] = [];
+  restaurantId!: number;
+  pieChart!: Chart<'doughnut', number[], string>;
+  barChart!: Chart<'bar', number[], string>;
+  showLog:boolean = true;
+
+  constructor(
+    private salesService: SalesService,
+    private route: ActivatedRoute,
+    private location: Location
+  ) {} 
+
+  ngOnInit(): void {
+    this.restaurantId = +this.route.snapshot.paramMap.get('restaurantId')!;
+    this.sales = this.salesService.getSalesByRestaurant(this.restaurantId);
+    this.createDoughnutChart();
+    this.createBarChart();
+  }
+
+  goBack(): void {
+    this.location.back();
+  }
+
+  private createDoughnutChart(): void {
+    const itemSales = this.sales.reduce((acc, sale) => {
+      if (!acc[sale.itemName]) {
+        acc[sale.itemName] = 0;
+      }
+      acc[sale.itemName] += sale.quantity;
+      return acc;
+    }, {} as Record<string,number>);
+    
+    const labels = Object.keys(itemSales).map(itemId => itemId);
+    const data = Object.values(itemSales);
+
+    this.pieChart = new Chart('donut-chart', {
+      type: 'doughnut',
+      data: {
+        labels,
+        datasets: [{
+          data,
+          backgroundColor: labels.map(() => this.getRandomColor()),
+          hoverOffset: 30
+        }]
+      },
+      options: {
+        responsive: true,
+        maintainAspectRatio: false
+      }
+    });
+  }
+
+  private createBarChart(): void {
+    const dailySales = this.sales.reduce((acc, sale) => {
```

Comment:
This variable creation will automatically asign types, but can we add the type manually since if there is any issuess occuring with the data mapping it will help us identify easily. Could you do this for all the temporary variable creation

Reply:
resolved in all instance